src/com/aca/Functionaluty/MemoryGadget.java

```diff
@@ -0,0 +1,22 @@
+package com.aca.Functionaluty;
+
+public class MemoryGadget extends Gadget {
+    private int memorysize;
+
+    public MemoryGadget(String name,int duration, int price, int count, int memorysize) {
+        super(name,duration, price, count);
+        this.memorysize = memorysize;
+
+    }
+
+    public MemoryGadget(String row) {
+        super(row);
+        String splitDetails = ",";
```

Comment:
move this splittings from models, resolve values of parameters outside of code and use the first constructor